stan/math/opencl/kernels/tri_multiply.hpp

```diff
@@ -0,0 +1,155 @@
+#ifndef STAN_MATH_OPENCL_KERNELS_TRI_MULTIPLY_HPP
+#define STAN_MATH_OPENCL_KERNELS_TRI_MULTIPLY_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+static const char* tri_multiply_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * OpenCL Matrix multiplication of a triangular and rectangular matrix
+     *
+     * @param[in] A the lower/upper triangular matrix in matrix multiplication
+     * @param[in] B the right matrix in matrix multiplication
+     * @param[out] C the output matrix
+     * @param[in] M Number of rows for matrix A
+     * @param[in] N Number of rows for matrix B
+     * @param[in] K Number of cols for matrix A and number of rows for matrix B
+     * @param[in] lower_upper_A the triangularity of A (lower, upper or none)
```

Comment:
forgive me if we already went over this, but could we use a C enum here?

Reply:
Yes we could. I am not sure if we discussed this or not either :) 

We would have to put it in the helpers where we would do something like this:
```
char* helpers =
    R"(
  enum triangular_view{
      TRIANGULAR_VIEW_LOWER = LOWER,
      TRIANGULAR_VIEW_UPPER = UPPER,
      TRIANGULAR_VIEW_ENTIRE = ENTIRE
  }; 

... rest of helpers
```
The LOWER, UPPER, ENTIRE are defines that are set at kernel compile times based on the values in our `TriangularViewCL` enum. This was done so that if someone goes and changes the values in TriangularViewCL, the whole thing would still work. If you think this is a fine solution, then I can add this easily.

Sadly the STRINGIFY() doesnt want to work with enums :/

Reply:
Gotcha. importing the actual int values from C++ makes sense. Should we just do this:
```cpp
std::ostringstream enums;
enums << "enum triangular_view {\n";
enums << "    LOWER=" << TriangularViewCL::Lower << ";\n";
enums << "    UPPER=" << TriangularViewCL::Upper << ";\n";
enums << "    ENTIRE=" << TriangularViewCL::Entire << ";\n};";
```
Or possibly the `const char*` version with sprintf, that might even be easier to read. We have another enum we could do this way too.

This is assuming that this will help you simplify the code using it somewhat, and make the type signature communicate a little more. This is just based on me guessing, though, so if you think or see that it's not making it easier we can drop it.

Reply:
Yeah, that looks way better than my idea. I believe this is useful but would rather make the changes to the entire `opencl_kernels` code base at once if you agree with that. I can take care of this tomorrow.